### PR TITLE
fix(material-experimental/mdc-form-field): make sure fonts are loaded…

### PR DESCRIPTION
**Pull request by annieyw**
… before calculating notch width

The label width was being calculated before the fonts were loaded which caused the label to be cut off sometimes. By  calculating the width after `fonts.ready` resolves we can be sure that the label will be the correct size. 
Note that the workaround for IE is not perfect as depending on the connection speed the font still might not be loaded after the time out.

Before: 
9qHkqooP5ytmq42

After:
5XkdAvbCDJpEwM6

### CODE REVIEW

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 1]
```diff
@@ -340,7 +340,15 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
     this._updateFocusState();
     // Initial notch width update. This is needed in case the text-field label floats
     // on initialization, and renders inside of the notched outline.
-    this._refreshOutlineNotchWidth();
+    if (this._platform.TRIDENT) {
```
Comment (crisbeto) [inline src/material-experimental/mdc-form-field/form-field.ts:343]: Rather than looking at the browser, you should be able to detect the feature instead. E.g. `if (document.fonts && document.fonts.ready)`. Usually we only do browser detection as a last resort when feature detection isn't possible.

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 2]
```diff
@@ -340,7 +340,15 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
...
+      // FontFaceSet API is not supported in IE
+      setTimeout(() => this._refreshOutlineNotchWidth());
+    } else {
+      // Make sure fonts are loaded before calculating the width.
+      // FontFaceSet is technically still an experimental API at this time so the cast is needed
+      // as experimental API's aren't in the lib file.
+      (document as any).fonts.ready.then(() => this._refreshOutlineNotchWidth());
```
Comment (crisbeto) [inline src/material-experimental/mdc-form-field/form-field.ts:350]: Accessing the `document` directly like this will throw an error during server-side rendering. You should use the `DOCUMENT` injection token instead. You can see some examples by searching for `@Inject(DOCUMENT)`.

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 3]
```diff
@@ -340,7 +342,13 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
     this._updateFocusState();
     // Initial notch width update. This is needed in case the text-field label floats
     // on initialization, and renders inside of the notched outline.
-    this._refreshOutlineNotchWidth();
+    // Make sure fonts are loaded before calculating the width.
+    if (this._document.fonts && this._document.fonts.ready) {
```
Comment (mmalerba) [inline src/material-experimental/mdc-form-field/form-field.ts:346]: could shorten this to `if (this._document?.fonts?.ready)`

**Comment by annieyw** [comment]
* use `DOCUMENT` injection token 
* use feature detection instead of checking for browser

**Review by crisbeto — APPROVED** [review]
LGTM

**Comment by annieyw** [comment]
Add `markForCheck` because original change was causing prefilled form fields to have a large gap
6PamohuFcy4JD7k

**Comment by annieyw** [comment]
Some tweaks to make sure this change works in async tests as zone.js doesn't patch the FontFaceSet API and to account for really slow connections.
